File: 프로그래머스/3/43238. 입국심사/입국심사.java
import java.util.*;

class Solution {
    public long solution(int n, int[] times) {
        long left = 1;
        long right = (long) Arrays.stream(times).max().getAsInt() * n;
        long answer = right;
        
        while(left <= right){
            long mid = left + (right - left) / 2;
            
            long people = 0;
            for(int time : times){
                people += mid / time;
            }
            
            if(people >= n){
                answer = mid;
                right = mid - 1;
            }else{
                left = mid + 1;                
            }
            
        }
        
        return answer;
    }
}

File: 프로그래머스/3/43238. 입국심사/입국심사.kt
import java.util.*;

class Solution {
    fun solution(n: Int, times: IntArray): Long {
        var left : Long = 1
        var right : Long = times.maxOrNull()?.let { it * n.toLong() } ?: throw IllegalArgumentException()
        var answer = right

        while(left <= right) {
            var mid = left + (right - left) / 2

            var people : Long = 0

            for (t in times) {
                people += mid / t  
            }

            if(people >= n){
                answer = mid
                right = mid - 1
            }else{
                left = mid + 1
            }
        }

        return answer
    }
}